### DISCUSSION TITLE
Cannot override a specific dependancy using a preview trait

### DISCUSSION THREAD

**Discussion by wtpalexander** [pointfreeco/swift-dependencies#292] [Q&A] [answered]
I'm using a repository & protocol via dependencies. I'm wanting to override the dependancy to use a specific mocked version in my preview rather than the `previewValue`. Can anyone spot what I'm missing as it's currently just using the `previewValue` rather than what I suppose in the `dependency` preview trait?

```swift
protocol FooRepositoryProtocol { ... }

final class FooRepository: FooRepositoryProtocol { ... }

private enum FooRepositoryKey: DependencyKey, TestDependencyKey {
    static let liveValue: FooRepositoryProtocol = FooRepository()
    static let previewValue: FooRepositoryProtocol = MockFooRepository()
}

extension DependencyValues {
    var fooRepository: FooRepositoryProtocol {
        get { self[FooRepositoryKey.self] }
        set { self[FooRepositoryKey.self] = newValue }
    }
}

@available(iOS 18, *)
#Preview(traits: .dependency(\.fooRepository, .mock)) {
    NavigationStack {
        SearchForFriendsView()
    }
}

private extension FooRepositoryProtocol where Self == MockFooRepository {
    static var mock: MockFooRepository { MockFooRepository() }
}
```

Appreciate any thoughts!

**Comment by mbrandonw** [comment, marked as answer]
Hi @wtpalexander, is it possible to provide a compiling, self-contained example of the behavior you are seeing? I'm not seeing this when I try it out. Here's an example of overriding a depending in the preview:

<img width="1840" alt="Screenshot 2024-10-15 at 2 18 29 PM" src="https://github.com/user-attachments/assets/7f02b942-fce4-46d3-a73b-7fdea1720dd7">

And this is what happens when I do not override the dependency:

<img width="1840" alt="Screenshot 2024-10-15 at 2 18 52 PM" src="https://github.com/user-attachments/assets/b31e9e9e-9a1c-4d87-ab70-37de4400c107">

**Reply by wtpalexander** [reply]
Thanks @mbrandonw - really appreciate the quick response. Was just creating a minimum reproducible example (which I should have done first - apologies) and also see it working as expected. There must be something in my wider set up that is causing the issue. I'll look into it further there. Thanks again!
